Code/Framework/AzToolsFramework/Tests/Viewport/ViewportUiManagerTests.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Math/Transform.h>
+#include <AzCore/UnitTest/TestTypes.h>
+#include <AzFramework/Viewport/CameraState.h>
+#include <AzFramework/Viewport/ViewportScreen.h>
+#include <AzTest/AzTest.h>
+#include <AzToolsFramework/Viewport/ViewportMessages.h>
+#include <AzToolsFramework/ViewportUi/ViewportUiManager.h>
+
+namespace UnitTest
+{
+    using ViewportUiDisplay = AzToolsFramework::ViewportUi::Internal::ViewportUiDisplay;
+    using ViewportUiElementId = AzToolsFramework::ViewportUi::ViewportUiElementId;
+    using ButtonGroup = AzToolsFramework::ViewportUi::Internal::ButtonGroup;
+    using ButtonId = AzToolsFramework::ViewportUi::ButtonId;
+
+    // child class of ViewportUiManager which exposes the protected button group and viewport display
+    class ViewportUiManagerTestable : public AzToolsFramework::ViewportUi::ViewportUiManager
+    {
+    public:
+        ViewportUiManagerTestable() = default;
+        ~ViewportUiManagerTestable() override = default;
+
+        const AZStd::unordered_map<AzToolsFramework::ViewportUi::ClusterId, AZStd::shared_ptr<ButtonGroup>>& GetClusterMap()
+        {
+            return m_clusterButtonGroups;
+        }
+
+        ViewportUiDisplay* GetViewportUiDisplay()
+        {
+            return m_viewportUi.get();
+        }
+    };
+
+    class ViewportManagerWrapper
+    {
+    public:
+        void Create()
+        {
+            m_viewportManager = AZStd::make_unique<ViewportUiManagerTestable>();
+            m_viewportManager->ConnectViewportUiBus(AzToolsFramework::ViewportUi::DefaultViewportId);
+            m_mockRenderOverlay = AZStd::make_unique<QWidget>();
+            m_parentWidget = AZStd::make_unique<QWidget>();
+            m_viewportManager->InitializeViewportUi(m_parentWidget.get(), m_mockRenderOverlay.get());
+        }
+
+        void Destroy()
+        {
+            m_viewportManager->DisconnectViewportUiBus();
+            m_viewportManager.reset();
+            m_mockRenderOverlay.reset();
+            m_parentWidget.reset();
+        }
+
+        ViewportUiManagerTestable* GetViewportManager()
+        {
+            return m_viewportManager.get();
+        }
+
+        QWidget* GetMockRenderOverlay()
+        {
+            return m_mockRenderOverlay.get();
+        }
```

Comment:
Suggestion: Please move the implementations to the `ViewportUiManagerTests.cpp`

Reply:
I moved it to a header because `ComponentModeTests.h` was including it as a cpp and I wanted to to include `ComponentModeTests.h` somewhere else but it wasn't compiling

Reply:
It should work to do

`ViewportUiManagerTests.h`
````
        void Create();
        void Destroy();
````

`ViewportUiManagerTests.cpp`
````
        void ViewportManagerWrapper::Create()
        {
            m_viewportManager = AZStd::make_unique<ViewportUiManagerTestable>();
            m_viewportManager->ConnectViewportUiBus(AzToolsFramework::ViewportUi::DefaultViewportId);
            m_mockRenderOverlay = AZStd::make_unique<QWidget>();
            m_parentWidget = AZStd::make_unique<QWidget>();
            m_viewportManager->InitializeViewportUi(m_parentWidget.get(), m_mockRenderOverlay.get());
        }

        void ViewportManagerWrapper::Destroy()
        {
            m_viewportManager->DisconnectViewportUiBus();
            m_viewportManager.reset();
            m_mockRenderOverlay.reset();
            m_parentWidget.reset();
        }
````

If that doesn't work then it'd need to be investigated, there might be some other issue somewhere.

Reply:
oh my bad I misinterpreted it as move the whole thing back to the cpp, not the implementation!